src/components/avatar/Avatar.tsx

```diff
@@ -151,23 +154,24 @@ const Avatar = ({
     className,
   );
 
-  const enableTooltip = tooltip && typeof fullName === 'string' && fullName.length > 0;
+  const enableTooltip = tooltip && ((typeof fullName === 'string' && fullName.length > 0) || tooltipProps);
```

Comment:
Nice!, I didn't even think about adding the props also in the original case but it makes a lot of sense.
I have 1 tiny suggestion for a further improvement now though 😅 
This way when overwriting the default styles it won't show an empty tooltip if there's no fullName

```suggestion
  const enableTooltip = tooltip && ((typeof fullName === 'string' && fullName.length > 0) || tooltipProps?.tooltip);
```